app/src/main/java/space/taran/arknavigator/ui/fragments/GalleryFragment.kt

```diff
@@ -37,7 +36,7 @@ import space.taran.arknavigator.utils.*
 class GalleryFragment(
     private val index: ResourcesIndex,
     private val storage: TagsStorage,
-    private val resources: ResourcesList,
+    private val resources: MutableList<ResourceId>,
```

Comment:
To keep the data up to date in case of screen rotation, you just need to store it in the presenter.

`ResourceGripPresenter` does not know where to get resources, they are provided by the `TagSelectorPresenter` or `ResourcesPresenter`. Now both of them are using the `ResourceIndex`. Since both resource and gallery screens refer to the same `index` (object) and it is stored in the presenter, there is no need to worry about transferring changes between screens and data persistence in case of rotating screens.

But if the app is killed by the Android and the user returns to the app, then the data will be restored from the bundle (if we switch to them). The data from the bundle will most likely be no longer up-to-date and it will be necessary to check the list of transferred resources for changes.